tensor/dense_io.go

```diff
@@ -71,8 +73,21 @@ func (t *Dense) WriteNpy(w io.Writer) (err error) {
 	bw.Write([]byte(header))
 
 	bw.seq = 0
-	for i := 0; i < t.len(); i++ {
-		bw.w(t.Get(i))
+	if t.IsMasked() {
+		fillval := t.FillValue()
+		it := FlatMaskedIteratorFromDense(t)
+		runtime.SetFinalizer(it, destroyIterator)
```

Comment:
`runtime.SetFinalizer` should be set in `FlatMaskedIteratorFromDense`

See comment in that function.